packages/cadl-ranch-specs/http/payload/json-merge-patch/main.tsp

```diff
@@ -0,0 +1,219 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+
+using TypeSpec.Http;
+
+@doc("Test for merge-patch+json content-type")
+@scenarioService("/json-merge-patch")
+namespace Payload.JsonMergePatch;
+
+@doc("Details about a resource.")
+model Resource {
+  name: string;
+  description?: string;
+  map: Record<InnerModel>;
+  array?: InnerModel[];
+  intValue?: int32;
+  floatValue?: float32;
+}
+
+@doc("It is the model used by Resource model")
+model InnerModel {
+  name: string;
+  description?: string;
+}
+
+@scenario
+@scenarioDoc("""
+
+Expected input body:
+```json
+{
+  "name": "Madge",
+  "description": "desc",
+  "map": {
+    "key": {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  },
+  "array": [
+    {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  ],
+  "intValue": 1,
+  "floatValue": 1.1
+}
+```
+
+Expected response body:
+```json
+{
+  "name": "Madge",
+  "description": "desc",
+  "map": {
+    "key": {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  },
+  "array": [
+    {
+      "name": "InnerMadge",
+      "description": "innerDesc"
+    }
+  ],
+  "intValue": 1,
+  "floatValue": 1.1
+}
+```
+""")
+@doc("Test content-type: application/merge-patch+json with required body")
+@route("/create/resource")
+@put
+op createResource(@body body: Resource): Resource;
```

Comment:
Do you have any guidance saying a type could be in both put and patch?

Reply:
Azure may use the core template. But this would be a common case in non-azure. First create the resource, then update. I did not find official guidelines. But there is a discussion in Java PR: https://github.com/Azure/autorest.java/pull/2424#discussion_r1432216605

Reply:
Yeah I think for Azure, we shouldn't have both put and patch with the same model as stated in https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#rest-response-body-is-resource-schema. But for non-azure, I'm not sure.